### PR TITLE
IRIS-4236 Replace old sign out with new one

### PR DESCRIPTION
**Pull request by slayful**
## Links

* http://wikia-inc.atlassian.net/browse/IRIS-4236

### CODE REVIEW

File: server/app/facets/operations/signout.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import HttpStatus from 'http-status-codes';
+
+function getContext(request) {
+	const token = getAccessToken(request),
+		userId = getUserId(request);
+	return {
+		url: authUtils.getHeliosInternalUrl(`/token/${token}`),
+		options: {
+			headers: getInternalHeaders(request, {
+				'X-Wikia-UserId': userId,
```
Comment (pwojcik86) [inline server/app/facets/operations/signout.js]: The user's id is not needed for signout

File: server/app/facets/operations/signout.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+				'Content-type': 'application/json',
```
Comment (pwojcik86) [inline server/app/facets/operations/signout.js]: I don't think this is necessary, there's no content being sent

File: server/app/facets/operations/signout.js (hunk 1 above)
Comment (slayful) [inline server/app/facets/operations/signout.js]: Well it is, but it's not documented.

Comment (pwojcik86) [inline server/app/facets/operations/signout.js]: You can simply send the access token in the cookie or `X-Wikia-AccessToken`

File: front/main/app/components/sign-out.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import Ember from 'ember';
+import {track, trackActions} from 'common/utils/track';
+
+const {Component} = Ember;
+
+export default Component.extend(
+	{
+		trackingLabel: 'open-logout',
+		classNames: ['wds-sign-out'],
+		trackClick: Ember.K,
```
Comment (kvas-damian) [inline front/main/app/components/sign-out.js]: `Ember.K` is deprecated: https://github.com/emberjs/ember.js/issues/14746

File: front/main/app/styles/component/design-system/_notifications.scss [hunk 4]
```diff
@@ -15,3 +15,31 @@
 		}
 	}
 }
+
+// TODO to be moved to the Design System
+.wds-sign-out {
+	&__button {
+		border: none;
```
Comment (kvas-damian) [inline front/main/app/styles/component/design-system/_notifications.scss]: :abc:

File: front/main/app/styles/component/design-system/_notifications.scss [hunk 5]
```diff
@@ -15,3 +15,31 @@
 		}
 	}
 }
+
+// TODO to be moved to the Design System
```
Comment (kvas-damian) [inline front/main/app/styles/component/design-system/_notifications.scss]: Do we have a ticket for that? Can't we use existing DS buttons?

File: front/main/app/templates/components/sign-out.hbs [hunk 6]
```diff
@@ -0,0 +1,7 @@
+{{! Logout is a POST, so we render a button inside a form here}}
+<form method="POST" action="/logout">
```
Comment (kvas-damian) [inline front/main/app/templates/components/sign-out.hbs]: Probably we should update DS API to reflect thos changes: http://sandbox-xw2.www.wikia.com/api/v1/design-system/wikis/3035/en/

File: front/main/app/templates/components/sign-out.hbs [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+	{{input type="hidden" name="redirect" value="/d"}}
```
Comment (kvas-damian) [inline front/main/app/templates/components/sign-out.hbs]: Probably we should consider implementing more smart solution than always redirecting to `/d`

File: front/main/app/templates/components/sign-out.hbs [hunk 8]
```diff
@@ -0,0 +1,7 @@
...
+	<button type="submit" class="wds-sign-out__button">
+		{{i18n 'global-navigation-user-sign-out' ns='design-system'}}
```
Comment (kvas-damian) [inline front/main/app/templates/components/sign-out.hbs]: message key should be provided by DS API

File: front/main/app/templates/components/sign-out.hbs [hunk 9]
```diff
@@ -0,0 +1,7 @@
+{{! Logout is a POST, so we render a button inside a form here}}
+<form method="POST" action="/logout">
+	{{input type="hidden" name="redirect" value="/d"}}
+	<button type="submit" class="wds-sign-out__button">
```
Comment (kvas-damian) [inline front/main/app/templates/components/sign-out.hbs]: It looks that we removed tracking on this component. Is it intended?

File: front/main/app/styles/component/design-system/_notifications.scss (hunk 5 above)
Comment (slayful) [inline front/main/app/styles/component/design-system/_notifications.scss]: Will do as part of https://wikia-inc.atlassian.net/browse/IRIS-3991
It's a button that's supposed to look like a regular link (same as current logouts). I don't think we've got things like that in Design System.

File: front/main/app/templates/components/sign-out.hbs (hunk 6 above)
Comment (slayful) [inline front/main/app/templates/components/sign-out.hbs]: https://wikia-inc.atlassian.net/browse/IRIS-4249

File: front/main/app/templates/components/sign-out.hbs (hunk 7 above)
Comment (slayful) [inline front/main/app/templates/components/sign-out.hbs]: That's how it currently works in Discussion. For App I'll put the current URL there, but for SPA it's a little bit more tricky and no one currently cares.

File: front/main/app/templates/components/sign-out.hbs (hunk 9 above)
Comment (slayful) [inline front/main/app/templates/components/sign-out.hbs]: Re-added

File: front/main/app/styles/component/design-system/_notifications.scss (hunk 5 above)
Comment (kvas-damian) [inline front/main/app/styles/component/design-system/_notifications.scss]: what about http://wikiadesignsystem.com/#/components/buttons text buttons?

**Pull request merged**